src/java.base/share/classes/java/text/SimpleDateFormat.java

```diff
@@ -1024,15 +1031,15 @@ private StringBuffer format(Date date, StringBuffer toAppendTo,
      */
     @Override
     public AttributedCharacterIterator formatToCharacterIterator(Object obj) {
-        StringBuffer sb = new StringBuffer();
+        StringBuilder sb = new StringBuilder();
```

Comment:
On second thought, we can just make this `StringBuf sb = StringBufFactory.of(new StringBuilder())` - or just add `StringBufFactory.of()` which defaults to create a StringBuilder-backed buf.